app/models/supplejack_api/top_metric.rb

```diff
@@ -38,14 +38,22 @@ def self.spawn
       dates.each do |date|
         METRICS.each do |metric|
           results = results(date, metric).each_with_object({}) do |record, hash|
-            hash[record.record_id] = record.send(metric)
+            hash[record.record_id.to_s] = record.send(metric)
```

Comment:
why are we calling `.to_s` here? aren't the ids already `String`s

Reply:
No before they are saved they are integers, then once they are saved in Mongo they become strings, so if you try and merge them together and add the same values together for duplicate keys, they don't match.